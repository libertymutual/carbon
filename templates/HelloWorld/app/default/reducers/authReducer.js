import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
  isLoggedIn: false,
  username: "",
  password: "",
  authUrl: "",
  clientId: "",
  clientSecret: "",
  configName: "",
  validatorId: "",
}

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function authReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.LOGIN:
      return { ...state, isLoggedIn: true };
    case ActionTypes.LOGOUT:
      return { ...state, isLoggedIn: false };
    case ActionTypes.AUTH_UPDATE_AUTH_URL:
      return { ...state, authUrl: action.authUrl };
    case ActionTypes.AUTH_UPDATE_CLIENT_ID:
      return { ...state, clientId: action.clientId };
    case ActionTypes.AUTH_UPDATE_CLIENT_SECRET:
      return { ...state, clientSecret: action.clientSecret };
    case ActionTypes.AUTH_UPDATE_CONFIG_NAME:
      return { ...state, configName: action.configName };
    case ActionTypes.AUTH_UPDATE_PASSWORD:
      return { ...state, password: action.password };
    case ActionTypes.AUTH_UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.AUTH_UPDATE_VALIDATOR_ID:
      return { ...state, validatorId: action.validatorId };
    default:
      return state;
  }
}
