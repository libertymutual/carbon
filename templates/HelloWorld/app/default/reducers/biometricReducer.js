import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = {
    username: "TestUsername",
    password: "TestPassword",
    credentialsSaved: false
};

/**
 * Reducer handling auth related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function biometricReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.UPDATE_USERNAME:
      return { ...state, username: action.username };
    case ActionTypes.UPDATE_PASSWORD:
      return { ...state, password: action.password };  
    case ActionTypes.SAVE_CREDENTIALS:
      return { ...state, credentialsSaved: true };
    case ActionTypes.DELETE_CREDENTIALS:
      return { ...state, credentialsSaved: false };
    default:
      return state;
  }
}
