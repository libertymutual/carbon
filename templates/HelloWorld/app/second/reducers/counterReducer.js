import * as ActionTypes from '../actions/ActionTypes';

/**
 * Initial Reducer State
 */
export const initialState = 0;

/**
 * Reducer handling counter related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function countReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.INCREMENT:
      return state + 1;
    case ActionTypes.DECREMENT:
      return state - 1;
    default:
      return state;

  }
}
