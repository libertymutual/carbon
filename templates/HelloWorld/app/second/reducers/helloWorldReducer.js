import * as ActionTypes from '../actions/ActionTypes';
import * as Constants from '../constants/Constants';

/**
 * Initial Reducer State
 */
export const initialState = {
  state: Constants.HELLO,
};

/**
 * Reducer handling hello world related state
 * @param {Object} state The current reducer state
 * @param {Object} action The action event that has been triggered
 * @returns {Object} The updated state
 */
export default function helloWorldReducer(state = initialState, action) {
  switch (action.type) {
    case ActionTypes.HELLO:
      return { ...state, state: Constants.HELLO };
    case ActionTypes.GOODBYE:
      return { ...state, state: Constants.GOODBYE };
    default:
      return state;
  }
}
