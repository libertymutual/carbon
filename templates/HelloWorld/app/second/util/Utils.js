const { EventManager } = require('NativeModules');

/**
 * Collection of static util functions
 */
const utils = {

  /**
   * Gets the app version as specified in the version attribute of package.json
   * @returns {String} The current version of the app
   */
  getAppVersion() {
    return process.env.npm_package_version || '1.0.0';
  },

  /**
   * Gets the app name as specified in the name attribute of package.json
   * @returns {String} The name of the app
   */
  getAppName() {
    return process.env.npm_package_name || '';
  },

  /**
   * Gets the app description as specified in the description attribute of package.json
   * @returns {String} The description of the app
   */
  getAppDescription() {
    return process.env.npm_package_description || '';
  },

  /**
   * Navigates to another DSS App Platform Feature
   * @param featureName The name of the feature to navigate to
   * @param options Navigation options
   */
  navigateToFeature(featureName, options) {
    // e.g. EventManager.navigateFromReactNative
    // ('Second', {moduleName: 'Second', initialProperties: {'hello': 'world'}});
    EventManager.navigateFromReactNative(featureName, options);
  },

};

export default utils;
