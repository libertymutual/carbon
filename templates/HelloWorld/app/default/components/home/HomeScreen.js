import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as Constants from '../../constants/Constants';
import * as PresentationConstants from '../../constants/PresentationConstants';
import * as NavigationTargets from '../../constants/NavigationTargets';
import AuthActions from '../../actions/AuthActions';
import Utils from '../../util/Utils';
const { EventManager } = require('NativeModules');

/**
 * Home Screen
 * @extends {React.Component}
 */
class HomeScreen extends React.Component {
  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    const isLoggedIn = this.props.auth.isLoggedIn;
    const loginButtonMessage = isLoggedIn ? PresentationConstants.HOMESCREEN_BUTTON_LABEL_LOGOUT
      : PresentationConstants.HOMESCREEN_BUTTON_LABEL_LOGIN;

    return (<ScrollView>
      <Text style={styles.homeScreenHeader}>{PresentationConstants.HOMESCREEN_WELCOME}</Text>
      <Text style={styles.homeScreenText}>{PresentationConstants.HOMESCREEN_INTRO}</Text>
      {isLoggedIn ? <Button
        onPress={this._navigateToProfileScreen}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_PROFILE}
      /> : null}
      <Button
        onPress={this._navigateToContactScreen}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_CONTACT_US}
      />
      <Button
        onPress={this._navigateToSecondFeature}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_OTHER_FEATURE}
      />
      <Button
        onPress={this._navigateToWebFeature}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_WEB_FEATURE}
      />
      <Button
        onPress={this._navigateToLoginScreen}
        title={loginButtonMessage}
      />
      <Button
        onPress={this._navigateToAboutScreen}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_ABOUT}
      />
      <Button
        onPress={this._navigateToBiometricScreen}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_BIOMETRIC}
      />
      <Button
        onPress={this._getBuildType}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_BUILD_TYPE}
      />
    </ScrollView>);
  }

  /**
   * Navigates to the Profile Screen
   * @private
   */
  _navigateToProfileScreen = () => {
    this.props.navigation.navigate(NavigationTargets.PROFILE);
  }

  /**
   * Navigates to the Contact Screen
   * @private
   */
  _navigateToContactScreen = () => {
    this.props.navigation.navigate(NavigationTargets.CONTACT);
  }

  /**
   * Navigates to the About Screen
   * @private
   */
  _navigateToAboutScreen = () => {
    this.props.navigation.navigate(NavigationTargets.ABOUT);
  }

  /**
   * Navigates to the Biometric Screen
   * @private
   */
  _navigateToBiometricScreen = () => {
    this.props.navigation.navigate(NavigationTargets.BIOMETRIC);
  }

    /**
   * Navigates to the Login Screen
   * @private
   */
  _navigateToLoginScreen = () => {
    this.props.navigation.navigate(NavigationTargets.LOGIN);
  }

    /**
   * Navigates to the Login Screen
   * @private
   */
  _navigateToHomeScreen = () => {
    this.props.navigation.navigate(NavigationTargets.HOME);
  }


  /**
   * Gets the current build type
   * @private
   */
  _getBuildType = () => {
    EventManager.getBuildType().then(buildType => {
      alert("Build Type: " + buildType)
    })
  }

  /**
   * Navigates to the second feature
   * @private
   */
  _navigateToSecondFeature = () => {
    Utils.navigateToFeature(Constants.EXTERNAL_FEATURE_SECOND, {
      moduleName: Constants.EXTERNAL_FEATURE_MODULE_SECOND,
      initialProperties: { hello: 'world' },
    });
  }

  /**
   * Navigates to the web feature
   * @private
   */
  _navigateToWebFeature() {
    Utils.navigateToFeature(Constants.EXTERNAL_FEATURE_WEB, {
      url: "index.html",
    });
  }
}

/**
 * Navigation Options
 */
HomeScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_HOME_SCREEN,
};

/**
 * Prop Types
 */
HomeScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
  authActions: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @param {Object} state Redux State Object
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps(state) {
  return {
    auth: state.auth,
  };
}

/**
 * Maps Dispatch Actions To Props
 * @param {function} Dispatch function
 * @returns {Object) Object representing the actions
 * that should be exposed to the component via props
 */
function mapDispatchToProps(dispatch) {
  return {
    authActions: bindActionCreators(AuthActions, dispatch),
  };
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  homeScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  homeScreenText: {
    margin: 14,
  },
});
