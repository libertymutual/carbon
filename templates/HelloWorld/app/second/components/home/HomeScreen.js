import React from 'react';
import PropTypes from 'prop-types';
import { Button, ScrollView, StyleSheet, Text } from 'react-native';
import { connect } from 'react-redux';
import * as Constants from '../../constants/Constants';
import * as PresentationConstants from '../../constants/PresentationConstants';
import * as StylingConstants from '../../constants/StylingConstants';
import * as NavigationTargets from '../../constants/NavigationTargets';
import Utils from '../../util/Utils';

/**
 * Home Screen
 * @extends {React.Component}
 */
class HomeScreen extends React.Component {
  /**
   * Renders the components
   * @override
   * @returns {JSX}
   */
  render() {
    return (<ScrollView style={styles.secondFeatureTheme}>
      <Text style={styles.homeScreenHeader}>{PresentationConstants.HOMESCREEN_WELCOME}</Text>
      <Text style={styles.homeScreenText}>{PresentationConstants.HOMESCREEN_INTRO}</Text>
      <Text
        style={styles.homeScreenText}
      >{PresentationConstants.HOMESCREEN_COUNTER_VALUE}{this.props.counter}</Text>
      <Button
        onPress={this._navigateToAboutScreen}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_ABOUT}
      />
      <Button
        onPress={this._navigateToDefaultFeature}
        title={PresentationConstants.HOMESCREEN_BUTTON_LABEL_BACK_TO_DEFAULT_FEATURE}
      />
    </ScrollView>);
  }

  /**
   * Navigates to the About Screen
   * @private
   */
  _navigateToAboutScreen = () => {
    this.props.navigation.navigate(NavigationTargets.ABOUT);
  }

  /**
   * Navigates to the default feature
   * @private
   */
  _navigateToDefaultFeature = () => {
    Utils.navigateToFeature(Constants.EXTERNAL_FEATURE_DEFAULT, {
      moduleName: Constants.EXTERNAL_FEATURE_MODULE_DEFAULT,
      initialProperties: { hello: 'world' },
    });
  }
}

/**
 * Navigation Options
 */
HomeScreen.navigationOptions = {
  title: PresentationConstants.PAGE_TITLE_HOME_SCREEN,
};

/**
 * Prop Types
 */
HomeScreen.propTypes = {
  counter: PropTypes.number.isRequired,
  navigation: PropTypes.object.isRequired,
};

/**
 * Maps State To Props
 * @param {Object} state Redux State Object
 * @returns {Object) Object representing the parts of state
 * that the component should subscribe to
 */
function mapStateToProps(state) {
  return {
    counter: state.counter,
  };
}

/**
 * Maps Dispatch Actions To Props
 * @returns {Object) Object representing the actions
 * that should be exposed to the component via props
 */
function mapDispatchToProps() {
  return {};
}

/**
 * Default export wrapped in a connect to expose state & actions via props
 */
export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);

/**
 * Component Styles
 */
const styles = StyleSheet.create({
  homeScreenHeader: {
    margin: 14,
    fontWeight: 'bold',
  },
  homeScreenText: {
    margin: 14,
  },
  secondFeatureTheme: {
    backgroundColor: StylingConstants.COLOR_LIGHTBLUE,
  },
});
