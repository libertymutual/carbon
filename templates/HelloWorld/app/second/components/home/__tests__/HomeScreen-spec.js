import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import HomeScreen from '../HomeScreen';
import { initialState } from '../../../reducers/counterReducer';
import Utils from '../../../util/Utils';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
let store;

describe('HomeScreen tests', () => {
  beforeEach(() => {
    // Sets up a new store
    store = mockStore({
      counter: initialState
    });
  });

  test('Home Screen renders correctly', () => {
    const tree = renderer.create(
      <HomeScreen store={store} navigation={{}}/>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
