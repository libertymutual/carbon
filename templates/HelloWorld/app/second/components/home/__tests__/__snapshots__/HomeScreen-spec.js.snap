// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen tests Home Screen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "lightblue",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontWeight": "bold",
          "margin": 14,
        }
      }
    >
      Welcome To The Second Feature Home Screen
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      This is an example of navigating to a completely separate React Native Feature

It also shows an example of sharing state between two separate React Native Features. The Counter Value below is the same counter value displayed/incremented/decremented on the Default Feature
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "margin": 14,
        }
      }
    >
      Counter Value:
      0
    </Text>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          About
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Back To Default Feature
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
